gwvolman/tasks.py

```diff
@@ -294,3 +290,63 @@ def publish(item_ids,
                        prov_info,
                        license_id)
     return res
+
+
+@girder_job(title='Create ad-hoc Tale')
+@app.task(bind=True)
+def create_adhoc_tale(self, imageId: str, dataId: List[str], spawn: bool):
+    """Create a Tale provided a url for an external data and an image Id."""
+    if spawn:
+        total = 4
+    else:
+        total = 3
+
+    self.job_manager.updateProgress(
+        message='Gathering basic info about the dataset', total=total,
+        current=1)
+    dataMap = self.girder_client.get(
+        '/repository/lookup', parameters={'dataId': json.dumps(dataId)})
+
+    self.job_manager.updateProgress(
+        message='Registering the dataset in Whole Tale', total=total,
+        current=2)
+    self.girder_client.post(
+        '/dataset/register', parameters={'dataMap': json.dumps(dataMap),
+                                         'copyToHome': True})
+
+    # Get resulting folder/item by name
+    user = self.girder_client.get('/user/me')
+    path = '/user/{}/Data/{}'.format(user['login'], dataMap[0]['name'])
+    resource = self.girder_client.get(
+        '/resource/lookup', parameters={'path': path})
+
+    payload = {
+        'authors': user['firstName'] + ' ' + user['lastName'],
+        'title': 'Ad-hoc Tale for ' + dataId[0],
```

Comment:
This is a hardcoded title, which means the title specified by the user is ignored.   We either need to drop the title box or use the value here.  Also, if the ID for the dataset is long (e.g., URI), it might not be useful here.

Reply:
Good catch. We should be taking an (arguably) optional Tale name as a parameter in the `/tale/createTaleFromDataset`  endpoint, pass it into `create_adhoc_tale`, and use it in ln 324.

The `Tale Name` box defaults to the package name and the user is free to modify it; it should be no problem passing it to the `createTaleFromDataset` endpoint.

Reply:
The problem is that we don't have the package name until we actually start the import process...
Maybe we shouldn't allow users to change the name? If it's "ad hoc tale" from dataset it probably doesn't matter, but what if it's going to be a serialized Tale under that url?

Reply:
I take the package name as a query parameter when someone lands on Whole Tale from a package, so it's on DataONE or Dataverse to let us know. The only place I'm using the package name is for the Tale name and as a placeholder for the folder that holds the data. If we have the option to pass in the Tale name I can pass the appropriate value to this from the UI.